core-processor/src/main/java/io/micronaut/aop/writer/AopProxyWriter.java

```diff
@@ -523,18 +580,28 @@ private void initConstructor(MethodElement constructor) {
             ParameterElement marker = ParameterElement.of(ClassElement.of("kotlin.jvm.internal.DefaultConstructorMarker"), "marker");
             realNewConstructorParameters.add(marker);
             superConstructorParametersBinding.add(Map.entry(marker, superConstructorParameterIndex));
-            this.newConstructorSignature = getConstructorDescriptor(realNewConstructorParameters);
-        } else {
-            this.newConstructorSignature = getConstructorDescriptor(newConstructorParameters);
-        }
-        this.newConstructor = MethodElement.of(
+
+            this.realConstructor = MethodElement.of(
                 proxyClass,
                 constructor.getAnnotationMetadata(),
                 proxyClass,
                 proxyClass,
                 "<init>",
-                newConstructorParameters.toArray(ZERO_PARAMETER_ELEMENTS)
+                realNewConstructorParameters.toArray(ZERO_PARAMETER_ELEMENTS)
+            );
+        }
+        this.newConstructor = MethodElement.of(
+            proxyClass,
+            constructor.getAnnotationMetadata(),
+            proxyClass,
+            proxyClass,
+            "<init>",
+            newConstructorParameters.toArray(ZERO_PARAMETER_ELEMENTS)
```

Comment:
would be good to add a `MethodElement.constructorOf` method that makes this simple

Reply:
This is a very unique case, I don't see a point of adding a new method